src/algebra/monoid_algebra/basic.lean

```diff
@@ -1224,6 +1237,21 @@ begin
     simp only [mul_one, to_add_of_add, smul_single', of_apply] },
 end
 
+--  `by convert monoid_algebra.monoid_ring_hom_map k f.to_multiplicative` appears to be the same
+--  mathematical definition, but is harder to use.
+/--  An additive homomorphism `f : G →+ H` induces a ring homomorphism
+`add_monoid_ring_hom_map f : add_monoid_algebra k G →+* add_monoid_algebra k H`. -/
+def add_monoid_ring_hom_map (k : Type*) [semiring k] {H F : Type*} [add_monoid G] [add_monoid H]
+  [add_monoid_hom_class F G H] (f : F) :
+  add_monoid_algebra k G →+* add_monoid_algebra k H :=
+{ map_one' := map_domain_one f,
+  map_mul' := λ x y, map_domain_mul f x y,
+  ..(finsupp.map_domain.add_monoid_hom f : monoid_algebra k G →+ monoid_algebra k H) }
+
+@[simp] lemma single_one_right_add_hom_apply (k : Type*) [semiring k] {H : Type*}
+  [add_zero_class G] [add_zero_class H] (f : G →+ H) (n : G) :
+  (of k H).comp f.to_multiplicative n = single (f n) 1 := rfl
```

Comment:
This lemma seems pointless to me, I think `dsimp` already knows how to prove it? If not, we're missing a simp lemma about `to_multiplicative`

Reply:
Ah, the lemmas are missing - I've added them in #13553

Reply:
I removed the remaining lemma.